File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/ZipAddress.js
/**
 * Information used for looking up jurisdictions using location names and/or zip codes.
 */

function ZipAddress() {

    this.CountryISO = "";
    this.County = "";
    this.Locality = "";
    this.State = "";
    this.ZipCode = "";
    this.ZipP4 = "";

    /**
     * Sets the three-character country ISO code.
     * @param countryISO String
     * @return Reference to this object for method chaining. 
     */
    this.countryISO = function (countryISO) {
        this.CountryISO = countryISO;
        return this;
    }

    /**
     * Three-character country ISO code.
     * @return String
     */
    this.getCountryISO = function () {
        return this.CountryISO;
    }

    /* Reference to this object for method chaining.*
     * Sets the three-character country ISO code.
     * @param countryISO String 
     */
    this.setCountryISO = function (countryISO) {
        this.CountryISO = countryISO;
    }

    /**
     * Sets the county name.
     * @param county String
     * @return Reference to this object for method chaining. 
     */
    this.county = function (county) {
        this.County = county;
        return this;
    }

    /**
     * County name.
     * @return String
     */
    this.getCounty = function () {
        return this.County;
    }

    /* Reference to this object for method chaining.*
     * Sets the county name.
     * @param county String 
     */
    this.setCounty = function (county) {
        this.County = county;
    }

    /**
     * Sets the locality name.
     * @param locality String
     * @return Reference to this object for method chaining. 
     */
    this.locality = function (locality) {
        this.Locality = locality;
        return this;
    }

    /**
     * Locality name.
     * @return String
     */
    this.getLocality = function () {
        return this.Locality;
    }

    /* Reference to this object for method chaining.*
     * Sets the locality name.
     * @param locality String 
     */
    this.setLocality = function (locality) {
        this.Locality = locality;
    }

    /**
     * Sets the two-character state abbreviation.
     * @param state String
     * @return Reference to this object for method chaining. 
     */
    this.state = function (state) {
        this.State = state;
        return this;
    }

    /**
     * Two-character state abbreviation.
     * @return String
     */
    this.getState = function () {
        return this.State;
    }

    /* Reference to this object for method chaining.*
     * Sets the two-character state abbreviation.
     * @param state String 
     */
    this.setState = function (state) {
        this.State = state;
    }

    /**
     * Sets the zip code for the location.
     * @param zipCode String
     * @return Reference to this object for method chaining. 
     */
    this.zipCode = function (zipCode) {
        this.ZipCode = zipCode;
        return this;
    }

    /**
     * Zip code for location.
     * @return String
     */
    this.getZipCode = function () {
        return this.ZipCode;
    }

    /* Reference to this object for method chaining.*
     * Sets the zip code for the location.
     * @param zipCode String 
     */
    this.setZipCode = function (zipCode) {
        this.ZipCode = zipCode;
    }

    /**
     * Sets the zip + 4 for the location.
     * @param zipP4 String
     * @return Reference to this object for method chaining. 
     */
    this.zipP4 = function (zipP4) {
        this.ZipP4 = zipP4;
        return this;
    }

    /**
     * Zip + 4 for location.
     * @return String
     */
    this.getZipP4 = function () {
        return this.ZipP4;
    }

    /* Reference to this object for method chaining.*
     * Sets the zip + 4 for the location.
     * @param zipP4 String 
     */
    this.setZipP4 = function (zipP4) {
        this.ZipP4 = zipP4;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/ZipAddress.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Information used for looking up jurisdictions using location names and/or zip codes.
 */
public class ZipAddress {

    @SerializedName("CountryISO")
    private String countryISO = null;

    @SerializedName("County")
    private String county = null;

    @SerializedName("Locality")
    private String locality = null;

    @SerializedName("State")
    private String state = null;

    @SerializedName("ZipCode")
    private String zipCode = null;

    @SerializedName("ZipP4")
    private String zipP4 = null;

    /**
     * Sets the three-character country ISO code.
     * @param countryISO
     * @return 
     */
    public ZipAddress countryISO(String countryISO) {
        this.countryISO = countryISO;
        return this;
    }

    /**
     * Three-character country ISO code.
     * @return countryISO
     */
    public String getCountryISO() {
        return countryISO;
    }

    /**
     * Sets the three-character country ISO code.
     * @param countryISO 
     */
    public void setCountryISO(String countryISO) {
        this.countryISO = countryISO;
    }

    /**
     * Sets the county name.
     * @param county
     * @return 
     */
    public ZipAddress county(String county) {
        this.county = county;
        return this;
    }

    /**
     * County name.
     * @return county
     */
    public String getCounty() {
        return county;
    }

    /**
     * Sets the county name.
     * @param county 
     */
    public void setCounty(String county) {
        this.county = county;
    }
    
    /**
     * Sets the locality name.
     * @param locality
     * @return 
     */
    public ZipAddress locality(String locality) {
        this.locality = locality;
        return this;
    }

    /**
     * Locality name.
     * @return locality
     */
    public String getLocality() {
        return locality;
    }

    /**
     * Sets the locality name.
     * @param locality 
     */
    public void setLocality(String locality) {
        this.locality = locality;
    }

    /**
     * Sets the two-character state abbreviation.
     * @param state
     * @return 
     */
    public ZipAddress state(String state) {
        this.state = state;
        return this;
    }

    /**
     * Two-character state abbreviation.
     * @return state
     */
    public String getState() {
        return state;
    }

    /**
     * Sets the two-character state abbreviation.
     * @param state 
     */
    public void setState(String state) {
        this.state = state;
    }
    
    /**
     * Sets the zip code for the location.
     * @param zipCode
     * @return 
     */
    public ZipAddress zipCode(String zipCode) {
        this.zipCode = zipCode;
        return this;
    }

    /**
     * Zip code for location.
     * @return zipCode
     */
    public String getZipCode() {
        return zipCode;
    }

    /**
     * Sets the zip code for the location.
     * @param zipCode 
     */
    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }
    
    /**
     * Sets the zip + 4 for the location.
     * @param zipP4
     * @return 
     */
    public ZipAddress zipP4(String zipP4) {
        this.zipP4 = zipP4;
        return this;
    }

    /**
     * Zip + 4 for location.
     * @return zipP4
     */
    public String getZipP4() {
        return zipP4;
    }

    /**
     * Sets the zip + 4 for the location.
     * @param zipP4 
     */
    public void setZipP4(String zipP4) {
        this.zipP4 = zipP4;
    }
}
